MIPs/mip-3.md

```diff
@@ -0,0 +1,117 @@
+---
+MIP: 1
+Title: Implement `wallet_swap` to support Dapps to swap
+Status: Draft
+Stability: Experimental
+discussions-to: https://github.com/MetaMask/metamask-improvement-proposals/discussions
+Author(s): Tomás Santos (@tommasini)
+Type: Community
+Created: 2023-11-10
+---
+
+## Summary
+
+This proposal introduces a new JSON RPC method `wallet_swap`` that allows dApps to request MetaMask to perform a token swap operation. The method accepts parameters for the source and destination tokens and initiates a token swap operation.
+
+## Motivation
+
+The motivation behind this MIP is to provide a standardized way for dApps to request token swaps from MetaMask, improving interoperability and user experience.
+
+# Usage Example
+
+```
+ethereum.request({
+  method: 'wallet_swap',
+  params: {
+    from: {
+      token_address: '0x1234567890abcdefABCDEF1234567890ABCDEF',
+      chainId: '0x1',
+      amount: '0xDE0B6B3A7640000',
+    },
+    to: {
+      token_address: '0xabcdef1234567890ABCDEF1234567890abcdef',
+      chainId: '0x1',
+    },
+  },
+});
```

Comment:
we should be using [`by-position` params](https://json-rpc.org/specification#parameter_structures). The only method that is `by-name` is wallet_watchAsset and that was a mistake that it went out like that.

```suggestion
ethereum.request({
  method: 'wallet_swap',
  params: [
    {
      from: {
        token_address: '0x1234567890abcdefABCDEF1234567890ABCDEF',
        chainId: '0x1',
        amount: '0xDE0B6B3A7640000',
      },
      to: {
        token_address: '0xabcdef1234567890ABCDEF1234567890abcdef',
        chainId: '0x1',
      },
    },
  ]
});
```

Reply:
Interesting! Will do this changes! Thank you so much!